--- MigrationTools-46/migrate_passwd.pl.orig	2009-10-29 15:17:51.000000000 +0100
+++ MigrationTools-46/migrate_passwd.pl	2009-11-11 10:47:07.000000000 +0100
@@ -202,25 +202,25 @@ sub dump_shadow_attributes
 	if ($pwd) {
 		print $HANDLE "userPassword: {crypt}$pwd\n";
 	}
-	if ($lastchg) {
+	if ($lastchg ne "") {
 		print $HANDLE "shadowLastChange: $lastchg\n";
 	}
-	if ($min) {
+	if ($min ne "") {
 		print $HANDLE "shadowMin: $min\n";
 	}
-	if ($max) {
+	if ($max ne "") {
 		print $HANDLE "shadowMax: $max\n";
 	}
-	if ($warn) {
+	if ($warn ne "") {
 		print $HANDLE "shadowWarning: $warn\n";
 	}
-	if ($inactive) {
+	if ($inactive ne "") {
 		print $HANDLE "shadowInactive: $inactive\n";
 	}
-	if ($expire) {
+	if ($expire ne "") {
 		print $HANDLE "shadowExpire: $expire\n";
 	}
-	if ($flag) {
+	if ($flag ne "") {
 		print $HANDLE "shadowFlag: $flag\n";
 	}
 }
