--- /usr/share/openldap/migration/migrate_automount.pl	2008-07-10 00:47:18.000000000 -0500
+++ ./migrate_automount.pl	2008-08-27 10:53:42.000000000 -0500
@@ -52,10 +52,10 @@
 }
 
 # setup the top level for this automounter map
-print $HANDLE "dn: nisMapName=$mapname,$NAMINGCONTEXT\n";
+print $HANDLE "dn: ou=$mapname,$NAMINGCONTEXT\n";
 print $HANDLE "objectClass: top\n";
-print $HANDLE "objectClass: nisMap\n";
-print $HANDLE "nisMapName: $mapname\n";
+print $HANDLE "objectClass: automountMap\n";
+print $HANDLE "ou: $mapname\n";
 print $HANDLE "\n";
 
 while(<INFILE>)
@@ -87,14 +87,13 @@
 	if ($key eq "*"){	
 		# since * isnt a valid attrib, replace it with "/" 
 		# which isnt a valid filename :->
-		print $HANDLE "dn: cn=/,nisMapName=$mapname,$NAMINGCONTEXT\n";
+		print $HANDLE "dn: cn=/,ou=$mapname,$NAMINGCONTEXT\n";
 	} else {
-		print $HANDLE "dn: cn=$key,nisMapName=$mapname,$NAMINGCONTEXT\n";
+		print $HANDLE "dn: cn=$key,ou=$mapname,$NAMINGCONTEXT\n";
 	}
-	print $HANDLE "objectClass: nisObject\n";
+	print $HANDLE "objectClass: automount\n";
 	print $HANDLE "cn: $key\n";
-	print $HANDLE "nisMapEntry: $value\n";
-	print $HANDLE "nisMapName: $mapname\n";
+	print $HANDLE "automountInformation: $value\n";
 	print $HANDLE "\n";
 }
 
