--- servers/slapd/syncrepl.c	16 Jun 2007 08:53:48 -0000	1.328
+++ servers/slapd/syncrepl.c	19 Jun 2007 09:49:50 -0000	1.329
@@ -320,6 +320,5 @@
 	BerElementBuffer berbuf;
 	BerElement *ber = (BerElement *)&berbuf;
 	LDAPControl c[2], *ctrls[3];
-	struct timeval timeout;
 	int rc;
 	int rhint;
@@ -405,12 +404,8 @@
 		ctrls[1] = NULL;
 	}
 
-	timeout.tv_sec = si->si_tlimit;
-	timeout.tv_usec = 0;
-
 	rc = ldap_search_ext( si->si_ld, base, scope, filter, attrs, attrsonly,
-		ctrls, NULL, si->si_tlimit > 0 ? &timeout : NULL,
-		si->si_slimit, &si->si_msgid );
+		ctrls, NULL, NULL, si->si_slimit, &si->si_msgid );
 	ber_free_buf( ber );
 	return rc;
 }
@@ -531,6 +531,7 @@
        op->o_ssf = ( op->o_sasl_ssf > op->o_tls_ssf )
                ?  op->o_sasl_ssf : op->o_tls_ssf;

+        ldap_set_option( si->si_ld, LDAP_OPT_TIMELIMIT, &si->si_tlimit );

        if ( BER_BVISNULL( &si->si_syncCookie.octet_str )) {
                /* get contextCSN shadow replica from database */
