--- openldap/servers/slapd/back-ldbm/back-ldbm.h
+++ openldap/servers/slapd/back-ldbm/back-ldbm.h
@@ -20,7 +20,7 @@
 #	define DEFAULT_DBCACHE_SIZE 100000
 #endif
 
-#define DEFAULT_DB_DIRECTORY	LDAP_RUNDIR LDAP_DIRSEP "openldap-ldbm"
+#define DEFAULT_DB_DIRECTORY	"/var/lib/ldap"
 #define DEFAULT_MODE		0600
 
 #define DN_BASE_PREFIX		SLAP_INDEX_EQUALITY_PREFIX
--- openldap/servers/slurpd/slurp.h
+++ openldap/servers/slurpd/slurp.h
@@ -40,7 +40,7 @@
 
 
 /* Default directory for slurpd's private copy of replication logs */
-#define	DEFAULT_SLURPD_REPLICA_DIR	LDAP_RUNDIR LDAP_DIRSEP "openldap-slurp"
+#define	DEFAULT_SLURPD_REPLICA_DIR	"/var/lib/ldap"
 
 /* Default name for slurpd's private copy of the replication log */
 #define	DEFAULT_SLURPD_REPLOGFILE	"slurpd.replog"
@@ -49,7 +49,7 @@
 #define	DEFAULT_SLURPD_STATUS_FILE	"slurpd.status"
 
 /* slurpd dump file - contents of rq struct are written here (debugging) */
-#define	SLURPD_DUMPFILE			LDAP_TMPDIR LDAP_DIRSEP "slurpd.dump"
+#define	SLURPD_DUMPFILE			DEFAULT_SLURPD_REPLICA_DIR "/slurpd.dump"
 
 /* default srvtab file.  Can be overridden */
 #define	SRVTAB				"/etc/srvtab"
--- openldap/servers/slapd/slapd.conf
+++ openldap/servers/slapd/slapd.conf
@@ -4,6 +4,11 @@
 # This file should NOT be world readable.
 #
 include		%SYSCONFDIR%/schema/core.schema
+include		%SYSCONFDIR%/schema/cosine.schema
+include		%SYSCONFDIR%/schema/inetorgperson.schema
+include		%SYSCONFDIR%/schema/nis.schema
+include		%SYSCONFDIR%/schema/redhat/autofs.schema
+include		%SYSCONFDIR%/schema/redhat/kerberosobject.schema
 
 # Define global ACLs to disable default read access.
 
@@ -11,8 +17,11 @@
 # service AND an understanding of referrals.
 #referral	ldap://root.openldap.org
 
-pidfile		%LOCALSTATEDIR%/slapd.pid
-argsfile	%LOCALSTATEDIR%/slapd.args
+#pidfile	%LOCALSTATEDIR%/slapd.pid
+#argsfile	%LOCALSTATEDIR%/slapd.args
+
+# Create a replication log in /var/lib/ldap for use by slurpd.
+#replogfile	/var/lib/ldap/master-slapd.replog
 
 # Load dynamic backend modules:
 # modulepath	%MODULEDIR%
@@ -20,7 +29,15 @@
 # moduleload	back_ldbm.la
 # moduleload	back_passwd.la
 # moduleload	back_shell.la
 
+#
+# The next three lines allow use of TLS for connections using a dummy test
+# certificate, but you should generate a proper certificate by changing to
+# /usr/share/ssl/certs, running "make slapd.pem", and fixing permissions on
+# slapd.pem so that the ldap user or group can read it.
+# TLSCertificateFile /usr/share/ssl/certs/slapd.pem
+# TLSCertificateKeyFile /usr/share/ssl/certs/slapd.pem
+# TLSCACertificateFile /usr/share/ssl/certs/ca-bundle.crt
 #
 # Sample Access Control
 #	Allow read access of root DSE
@@ -49,11 +66,17 @@
 rootdn		"cn=Manager,dc=my-domain,dc=com"
 #rootdn		"cn=Manager,o=My Organization Name,c=US"
 # Cleartext passwords, especially for the rootdn, should
-# be avoid.  See slappasswd(8) and slapd.conf(5) for details.
+# be avoided.  See slappasswd(8) and slapd.conf(5) for details.
 # Use of strong authentication encouraged.
-rootpw		secret
+# rootpw		secret
+# rootpw		{crypt}ijFYNcSNctBYg
 # The database directory MUST exist prior to running slapd AND 
 # should only be accessible by the slapd/tools. Mode 700 recommended.
-directory	%LOCALSTATEDIR%/openldap-ldbm
+directory	/var/lib/ldap
 # Indices to maintain
-index	objectClass	eq
+index	objectClass,uid,uidNumber,gidNumber,memberUid	eq
+index	cn,mail,surname,givenname			eq,subinitial
+# Replicas to which we should propagate changes
+#replica host=ldap-1.example.com:389 tls=yes
+#	bindmethod=sasl saslmech=GSSAPI
+#	authcId=host/ldap-master.example.com@EXAMPLE.COM
--- openldap/doc/man/man8/slurpd.8
+++ openldap/doc/man/man8/slurpd.8
@@ -112,7 +112,7 @@
 temporary files may contain sensitive information.
 This option allows you to specify the location of these temporary files. 
 The default is
-.BR LOCALSTATEDIR/openldap-slurp .
+.BR /var/lib/ldap .
 .TP
 .BI \-k " srvtab\-file"
 Specify the location of the kerberos srvtab file which contains keys
