--- openldap-2.2.13/servers/slapd/back-ldbm/alias.c.orig	2004-01-01 19:16:37.000000000 +0100
+++ openldap-2.2.13/servers/slapd/back-ldbm/alias.c	2007-05-31 13:58:38.000000000 +0200
@@ -98,7 +98,8 @@
 
 			/* check if aliasDN is a subordinate of any DN in our list */
 			if( dnlist_subordinate( dnlist, &aliasDN ) ) {
-				ch_free( aliasDN.bv_val );
+				/* Do not free the value - it's pointer do data in cache! */
+				/* ch_free( aliasDN.bv_val ); */
 				*matched = entry;
 				entry = NULL;
 				*err = LDAP_ALIAS_PROBLEM;
@@ -109,7 +110,8 @@
 			/* attempt to dereference alias */
 
 			newe = dn2entry_r( be, &aliasDN, &sup );
-			ch_free( aliasDN.bv_val );
+			/* Do not free the value - it's pointer do data in cache! */
+			/* ch_free( aliasDN.bv_val ); */
 
 			if( newe != NULL ) {
 				cache_return_entry_r(&li->li_cache, entry );
