Make libldap_r.so link with pthread.

--- openldap-2.3.34/libraries/libldap_r/Makefile.in.orig	2007-01-02 22:43:50.000000000 +0100
+++ openldap-2.3.34/libraries/libldap_r/Makefile.in	2007-04-24 10:27:53.000000000 +0200
@@ -56,7 +56,7 @@
 XXLIBS = $(SECURITY_LIBS) $(LUTIL_LIBS)
 XXXLIBS = $(LTHREAD_LIBS)
 NT_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
-UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
+UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS) $(LTHREAD_LIBS)
 
 .links : Makefile
 	@for i in $(XXSRCS); do \
