--- openldap-2.3.27/servers/slapd/sets.c	8 Sep 2006 19:20:26 -0000	1.29
+++ openldap-2.3.27/servers/slapd/sets.c	20 Dec 2006 04:41:14 -0000	1.30
@@ -265,10 +265,14 @@
 
 done:;
 	if ( !( op_flags & SLAP_SET_LREFARR ) && lset != NULL ) {
+		if ( !( op_flags & SLAP_SET_LREFVAL ))
+			cp->set_op->o_tmpfree( lset->bv_val, cp->set_op->o_tmpmemctx );
 		cp->set_op->o_tmpfree( lset, cp->set_op->o_tmpmemctx );
 	}
 
 	if ( !( op_flags & SLAP_SET_RREFARR ) && rset != NULL ) {
+		if ( !( op_flags & SLAP_SET_RREFVAL ))
+			cp->set_op->o_tmpfree( rset->bv_val, cp->set_op->o_tmpmemctx );
 		cp->set_op->o_tmpfree( rset, cp->set_op->o_tmpmemctx );
 	}
 
