--- openldap-2.3.43/servers/slapd/slapcommon.c.orig	2009-11-05 16:31:52.000000000 +0100
+++ openldap-2.3.43/servers/slapd/slapcommon.c	2009-11-05 16:32:12.000000000 +0100
@@ -72,7 +72,7 @@ usage( int tool, const char *progname )
 
 	case SLAPCAT:
 		options = " [-c]\n\t[-g] [-n databasenumber | -b suffix]"
-			" [-l ldiffile] [-a filter]\n";
+			" [-l ldiffile] [-s subtree-dn] [-a filter]\n";
 		break;
 
 	case SLAPDN:
