diff -rup openldap-2.1.30/libraries/libldap/tls.c openldap-2.1.30.fixed/libraries/libldap/tls.c
--- openldap-2.1.30/libraries/libldap/tls.c	2003-05-06 15:02:21.000000000 +0200
+++ openldap-2.1.30.fixed/libraries/libldap/tls.c	2010-06-22 15:52:23.793082351 +0200
@@ -963,7 +963,7 @@ ldap_pvt_tls_check_hostname( LDAP *ld, v
 	X509 *x;
 	const char *name;
 	char *ptr;
-	int ntype = IS_DNS;
+	int ntype = IS_DNS, nlen;
 #ifdef LDAP_PF_INET6
 	struct in6_addr addr;
 #else
@@ -977,6 +977,7 @@ ldap_pvt_tls_check_hostname( LDAP *ld, v
 	} else {
 		name = name_in;
 	}
+	nlen = strlen(name);
 
 	x = tls_get_cert((SSL *)s);
 	if (!x) {
@@ -1017,15 +1018,14 @@ ldap_pvt_tls_check_hostname( LDAP *ld, v
 		ex = X509_get_ext(x, i);
 		alt = X509V3_EXT_d2i(ex);
 		if (alt) {
-			int n, len1 = 0, len2 = 0;
+			int n, len2 = 0;
 			char *domain = NULL;
 			GENERAL_NAME *gn;
 
 			if (ntype == IS_DNS) {
-				len1 = strlen(name);
 				domain = strchr(name, '.');
 				if (domain) {
-					len2 = len1 - (domain-name);
+					len2 = nlen - (domain-name);
 				}
 			}
 			n = sk_GENERAL_NAME_num(alt);
@@ -1040,7 +1040,7 @@ ldap_pvt_tls_check_hostname( LDAP *ld, v
 					sl = ASN1_STRING_length(gn->d.ia5);
 
 					/* Is this an exact match? */
-					if ((len1 == sl) && !strncasecmp(name, sn, len1)) {
+					if ((nlen == sl) && !strncasecmp(name, sn, nlen)) {
 						break;
 					}
 
@@ -1086,13 +1086,28 @@ ldap_pvt_tls_check_hostname( LDAP *ld, v
 
 	if (ret != LDAP_SUCCESS) {
 		X509_NAME *xn;
-		char buf[2048];
+		X509_NAME_ENTRY *ne;
+		ASN1_OBJECT *obj;
+		ASN1_STRING *cn = NULL;
+		int navas;
+
+		/* find the last CN */
+		obj = OBJ_nid2obj( NID_commonName );
+		if ( !obj ) goto no_cn; /* should never happen */
 
 		xn = X509_get_subject_name(x);
+		navas = X509_NAME_entry_count( xn );
+		for ( i=navas-1; i>=0; i-- ) {
+			ne = X509_NAME_get_entry( xn, i );
+			if ( !OBJ_cmp( ne->object, obj )) {
+				cn = X509_NAME_ENTRY_get_data( ne );
+				break;
+			}
+		}
 
-		if( X509_NAME_get_text_by_NID( xn, NID_commonName,
-			buf, sizeof(buf)) == -1)
+		if( !cn )
 		{
+no_cn:
 #ifdef NEW_LOGGING
 			LDAP_LOG ( TRANSPORT, ERR, "ldap_pvt_tls_check_hostname: "
 				"TLS unable to get common name from peer certificate.\n", 
@@ -1104,21 +1119,24 @@ ldap_pvt_tls_check_hostname( LDAP *ld, v
 #endif
 			ld->ld_error = LDAP_STRDUP("TLS: unable to get CN from peer certificate");
 
-		} else if (strcasecmp(name, buf)) {
+		} else if ( cn->length == nlen &&
+			strncasecmp( name, (char *) cn->data, nlen ) == 0 ) {
+			ret = LDAP_SUCCESS;
+
+		} else {
 #ifdef NEW_LOGGING
 			LDAP_LOG ( TRANSPORT, ERR, "ldap_pvt_tls_check_hostname: "
 				"TLS hostname (%s) does not match "
-				"common name in certificate (%s).\n", name, buf, 0 );
+				"common name in certificate (%.*s).\n",
+				name, cn->length, cn->data );
 #else
 			Debug( LDAP_DEBUG_ANY, "TLS: hostname (%s) does not match "
-				"common name in certificate (%s).\n", 
-				name, buf, 0 );
+				"common name in certificate (%.*s).\n", 
+				name, cn->length, cn->data );
 #endif
 			ret = LDAP_CONNECT_ERROR;
 			ld->ld_error = LDAP_STRDUP("TLS: hostname does not match CN in peer certificate");
 
-		} else {
-			ret = LDAP_SUCCESS;
 		}
 	}
 	X509_free(x);
