commit dcf94e25f57c2f24936c3ce66d2350c391354da1
Author: Howard Chu <hyc@openldap.org>
Date:   Mon Mar 26 03:32:11 2012 -0700

    ITS#7207 check for ld->ld_defconn

diff --git a/libraries/libldap/cyrus.c b/libraries/libldap/cyrus.c
index 0248101..2475a87 100644
--- a/libraries/libldap/cyrus.c
+++ b/libraries/libldap/cyrus.c
@@ -419,7 +419,7 @@ ldap_int_sasl_bind(
 		LDAP_MUTEX_LOCK( &ld->ld_conn_mutex );
 		ber_sockbuf_ctrl( ld->ld_sb, LBER_SB_OPT_GET_FD, &sd );
 
-		if ( sd == AC_SOCKET_INVALID ) {
+		if ( sd == AC_SOCKET_INVALID || !ld->ld_defconn ) {
 			/* not connected yet */
 
 			rc = ldap_open_defconn( ld );
