--- openldap-2.2.13/libraries/libldap/url.c.orig	2007-05-30 15:52:16.000000000 +0200
+++ openldap-2.2.13/libraries/libldap/url.c	2007-05-30 15:52:48.000000000 +0200
@@ -547,6 +547,12 @@
 		}
 
 		ludp->lud_port = atoi( q );
+                if ( (ludp->lud_port < 0) || (ludp->lud_port > 0xffff) ) {
+                        /* Invalid port number was assigned */
+                        LDAP_FREE( url );
+                        ldap_free_urldesc( ludp );
+                        return LDAP_URL_ERR_BADURL;
+		}
 	}
 
 	ldap_pvt_hex_unescape( url );
