--- servers/slapd/sets.c	2004/01/01 18:15:32	1.19
+++ servers/slapd/sets.c	2004/08/06 14:14:05	1.20
@@ -209,8 +209,8 @@
 	SetCookie *cp, struct berval *fbv,
 	struct berval *user, struct berval *this, BerVarray *results)
 {
-#define IS_SET(x)	( (long)(x) >= 256 )
-#define IS_OP(x)	( (long)(x) < 256 )
+#define IS_SET(x)        ( (unsigned long)(x) >= 256 )
+#define IS_OP(x)        ( (unsigned long)(x) < 256 )
 #define SF_ERROR(x)	do { rc = -1; goto _error; } while (0)
 #define SF_TOP()	( (BerVarray)( (stp < 0) ? 0 : stack[stp] ) )
 #define SF_POP()	( (BerVarray)( (stp < 0) ? 0 : stack[stp--] ) )
@@ -220,7 +220,7 @@
 	} while (0)
 
 	BerVarray set, lset;
-	BerVarray stack[64];
+	BerVarray stack[64] = { 0 };
 	int len, op, rc, stp;
	char c, *filter = fbv->bv_val;

