--- openldap-2.1.17/libraries/libldap/init.c	2003-04-30 22:12:48.000000000 -0400
+++ openldap-2.1.17/libraries/libldap/init.c	2003-04-30 22:12:58.000000000 -0400
@@ -544,6 +544,7 @@
 #endif
 
 	openldap_ldap_init_w_sysconf(LDAP_CONF_FILE);
+	if( ( getuid() == geteuid() ) && ( getgid() == getegid() ) ) {
 	openldap_ldap_init_w_userconf(LDAP_USERRC_FILE);
 
 	{
@@ -597,6 +598,7 @@
 	}
 
 	openldap_ldap_init_w_env(gopts, NULL);
+	}
 
 	ldap_int_sasl_init();
 }
