--- openldap-2.0.12/libraries/libldap/test.c	Sat Jul 21 15:01:40 2001
+++ openldap-2.0.12/libraries/libldap/test.c	Fri Aug 31 15:42:30 2001
@@ -250,7 +250,7 @@
 #endif /* LDAP_API_FEATURE_X_OPENLDAP_V2_KBIND */
 
 		getline( dn, sizeof(dn), stdin, "re-bind dn? " );
-		strcat( dn, dnsuffix );
+		strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 
 	if ( authmethod == LDAP_AUTH_SIMPLE && dn[0] != '\0' ) {
 			getline( passwd, sizeof(passwd), stdin,
@@ -372,7 +372,7 @@
 			switch ( command2 ) {
 			case 'd':	/* add */
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				if ( (attrs = get_modlist( NULL, "attr? ",
 				    "value? " )) == NULL )
 					break;
@@ -405,7 +405,7 @@
 			method = LDAP_AUTH_SIMPLE;
 #endif /* LDAP_API_FEATURE_X_OPENLDAP_V2_KBIND */
 			getline( dn, sizeof(dn), stdin, "dn? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) -1 );
 
 			if ( method == LDAP_AUTH_SIMPLE && dn[0] != '\0' )
 				getline( passwd, sizeof(passwd), stdin,
@@ -435,7 +435,7 @@
 			method = LDAP_AUTH_SIMPLE;
 #endif /* LDAP_API_FEATURE_X_OPENLDAP_V2_KBIND */
 			getline( dn, sizeof(dn), stdin, "dn? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 
 			if ( dn[0] != '\0' )
 				getline( passwd, sizeof(passwd), stdin,
@@ -455,7 +455,7 @@
 
 		case 'c':	/* compare */
 			getline( dn, sizeof(dn), stdin, "dn? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 			getline( attr, sizeof(attr), stdin, "attr? " );
 			getline( value, sizeof(value), stdin, "value? " );
 
@@ -500,7 +500,7 @@
 		case 'm':	/* modify or modifyrdn */
 			if ( strncmp( line, "modify", 4 ) == 0 ) {
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				if ( (mods = get_modlist(
 				    "mod (0=>add, 1=>delete, 2=>replace -1=>done)? ",
 				    "attribute type? ", "attribute value? " ))
@@ -513,7 +513,7 @@
 					    id );
 			} else if ( strncmp( line, "modrdn", 4 ) == 0 ) {
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				getline( rdn, sizeof(rdn), stdin, "newrdn? " );
 				if ( (id = ldap_modrdn( ld, dn, rdn )) == -1 )
 					ldap_perror( ld, "ldap_modrdn" );
@@ -558,7 +558,7 @@
 
 			case 'm':	/* remove */
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				if ( (id = ldap_delete( ld, dn )) == -1 )
 					ldap_perror( ld, "ldap_delete" );
 				else
@@ -574,7 +574,7 @@
 
 		case 's':	/* search */
 			getline( dn, sizeof(dn), stdin, "searchbase? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 			getline( line, sizeof(line), stdin,
 			    "scope (0=Base, 1=One Level, 2=Subtree)? " );
 			scope = atoi( line );
