--- openldap-2.0.11/build/top.mk	Thu May 10 22:47:00 2001
+++ openldap-2.0.11/build/top.mk	Fri Jul  6 00:30:37 2001
@@ -97,7 +97,7 @@
 INCLUDEDIR= $(top_srcdir)/include
 LDAP_INCPATH= -I$(LDAP_INCDIR) -I$(INCLUDEDIR)
 LDAP_LIBADIR= $(top_builddir)/libraries
-LDAP_LIBPATH= -L$(LDAP_LIBADIR)
+LDAP_LIBPATH= -L. -L$(LDAP_LIBADIR)/libavl -L$(LDAP_LIBADIR)/liblber -L$(LDAP_LIBADIR)/liblber/.libs -L$(LDAP_LIBADIR)/libldap -L$(LDAP_LIBADIR)/libldap/.libs -L$(LDAP_LIBADIR)/libldap_r -L$(LDAP_LIBADIR)/libldap_r/.libs -L$(LDAP_LIBADIR)/libldbm -L$(LDAP_LIBADIR)/libldif -L$(LDAP_LIBADIR)/liblunicode -L$(LDAP_LIBADIR)/liblutil
 
 LUTIL_LIBS = @LUTIL_LIBS@
 LDIF_LIBS = @LDIF_LIBS@
--- openldap-2.0.11/build/mod.mk	Wed Oct  4 11:21:30 2000
+++ openldap-2.0.11/build/mod.mk	Fri Jul  6 00:30:37 2001
@@ -10,7 +10,7 @@
 LIBRARY = $(LIBBASE).la
 LIBSTAT = lib$(LIBBASE).a
 
-LTFLAGS = --only-$(LINKAGE)
+#LTFLAGS = --only-$(LINKAGE)
 
 COMPILE = $(LIBTOOL) $(LTFLAGS) --mode=compile $(CC) $(CFLAGS) $(MODDEFS) -c
 LTLIBLINK = $(LIBTOOL) $(LTFLAGS) --mode=link $(CC) -rpath $(moduledir) \
