Resolves: #910241

commit 329e7937e69d52809431273152d7a19039283f66
Author: Howard Chu <hyc@openldap.org>
Date:   Fri Jun 10 01:44:30 2011 -0700

    ITS#6892 shortcut for non-replicated ops

diff --git a/servers/slapd/overlays/syncprov.c b/servers/slapd/overlays/syncprov.c
index a058e19..6f10923 100644
--- a/servers/slapd/overlays/syncprov.c
+++ b/servers/slapd/overlays/syncprov.c
@@ -2048,6 +2048,9 @@ syncprov_op_mod( Operation *op, SlapReply *rs )
 			opc->rsid = scook->sid;
 	}
 
+	if ( op->o_dont_replicate )
+		return SLAP_CB_CONTINUE;
+
 	/* If there are active persistent searches, lock this operation.
 	 * See seqmod.c for the locking logic on its own.
 	 */
