--- openldap-2.0.27/servers/slapd/acl.c	2002-08-26 07:39:26.000000000 -0400
+++ acl.c.patched	2006-08-04 14:07:27.000000000 -0400
@@ -405,6 +405,34 @@
 			 * user is bound as somebody in the same namespace as
 			 * the entry, OR the given dn matches the dn pattern
 			 */
+			if ( b->a_dn_self ) {
+			    const char *dummy;
+			    int rc;
+			    int match;
+			    struct berval bvTmp;
+
+			    /* if ( desc->ad_type->sat_syntax
+ != slap_schema.si_syn_distinguishedName
+ && !is_at_syntax( desc->ad_type, SLAPD_NAMEUID_SYNTAX )) {
+				Debug( LDAP_DEBUG_ACL, "acl_mask: Syntax fails\n", 0, 0, 0);
+				continue;
+			    } */
+			    /* check if the target is an attribute. */
+			    if ( val == NULL ) {
+				continue;
+			    }
+			    /* target is attribute, check if the attribute value
+			       is the op dn. */
+			    match = 0;
+			    bvTmp.bv_len = strlen(op->o_ndn);
+			    bvTmp.bv_val = op->o_ndn;
+			    rc = value_match( &match, desc,
+ desc->ad_type->sat_equality, 0, val, &bvTmp, &dummy );
+			    /* on match error or no match, fail the ACL clause */
+			    if ( rc != LDAP_SUCCESS || match != 0 ) {
+				continue;
+			    }
+			}
 			if ( strcmp( b->a_dn_pat, "anonymous" ) == 0 ) {
 				if (op->o_ndn != NULL && op->o_ndn[0] != '\0' ) {
 					continue;
