--- openldap-2.0.12/clients/ud/edit.c	Tue Jul  4 13:58:45 2000
+++ openldap-2.0.12/clients/ud/edit.c	Fri Aug 31 15:42:30 2001
@@ -129,6 +129,7 @@
 load_editor( void )
 {
 	FILE *fp;
+	int fd;
 	char *cp, *editor = UD_DEFAULT_EDITOR;
 	int tmpfd;
 #ifndef HAVE_SPAWNLP
--- openldap-2.0.12/clients/ud/group.c	Thu Sep 14 20:49:27 2000
+++ openldap-2.0.12/clients/ud/group.c	Fri Aug 31 15:42:30 2001
@@ -752,6 +752,7 @@
 	 *  are very straight-forward, but the last case requires a little
 	 *  extra care and work.
 	 */
+	memset( attrtype, '\0', sizeof(attrtype) );
 	if (verbose) {
 		printf("\n");
 		if ( !isatty( 1 ))
@@ -884,13 +885,13 @@
 			}
 			new_value = s;
 			strcpy(attrtype, "rfc822");
-			strcat(attrtype, attrlist[offset].quipu_name);
+			strncat(attrtype, attrlist[offset].quipu_name, sizeof(attrtype) - strlen(attrtype) - 1);
 			mod.mod_type = attrtype;
 		}
 		else {
 			new_value = s;
 			strcpy(attrtype, "rfc822");
-			strcat(attrtype, attrlist[offset].quipu_name);
+			strncat(attrtype, attrlist[offset].quipu_name, sizeof(attrtype) - strlen(attrtype) - 1);
 			mod.mod_type = attrtype;
 		}
 
@@ -954,7 +955,7 @@
 				mod.mod_type = mod.mod_type + 6;
 			else {
 				strcpy(attrtype, "rfc822");
-				strcat(attrtype, mod.mod_type);
+				strncat(attrtype, mod.mod_type, sizeof(attrtype) - strlen(attrtype) - 1);
 				mod.mod_type = attrtype;
 			}
 			mods[0] = &mod;
--- openldap-2.0.12/clients/ud/mod.c	Tue Jul  4 13:58:45 2000
+++ openldap-2.0.12/clients/ud/mod.c	Fri Aug 31 15:42:30 2001
@@ -385,6 +385,7 @@
 	if (debug & D_TRACE)
 		printf("->get_value(%s, %s)\n", id, prompt);
 #endif
+	memset(line, '\0', sizeof(line));
 	/* if this is a URL, have another routine handle this */
 	if (!strcmp(id, "labeledURL"))
 		return(get_URL());
@@ -487,7 +488,7 @@
 				return(NULL);
 			}
 			tmp = ldap_get_dn(ld, elmp);
-			strcpy(buffer, tmp);
+			strncpy(buffer, tmp, sizeof(buffer) - 1);
 			ldap_memfree(tmp);
 			(void) ldap_msgfree(lmp);
 			break;
@@ -509,8 +510,10 @@
 		 *  are done.
 		 */
 		if (count++ > 1)
-			(void) strcat(buffer, " $ ");
-		(void) strcat(buffer, line);
+			(void) strncat(buffer, " $ ",
+				       sizeof(buffer) - strlen(buffer) - 1);
+		(void) strncat(buffer, line,
+			       sizeof(buffer) - strlen(buffer) - 1);
 		if (!multiline)
 			break;
 		if ((count > MAX_LINES) && (!strcmp(id, "postalAddress") || !strcmp(id, "homePostalAddress"))) {
@@ -616,6 +619,7 @@
 	mod.mod_op = LDAP_MOD_REPLACE;
 	mod.mod_type = "noBatchUpdates";
 	mod.mod_values = newsetting;
+	memset(value, '\0', sizeof(value));
 	/* explain what the implications are */
 	if (verbose) {
 		printf("\n  By default, updates that are received from the Personnel\n");
--- openldap-2.0.12/clients/fax500/faxtotpc.c	Tue Jun 13 13:57:01 2000
+++ openldap-2.0.12/clients/fax500/faxtotpc.c	Fri Aug 31 15:42:30 2001
@@ -185,8 +185,11 @@
 	char ibuf[255];
 	char obuf[255];
 
+	memset(ibuf, '\0', sizeof(ibuf));
+	memset(obuf, '\0', sizeof(obuf));
+
 	/* nuke spaces */
-	strcpy(ibuf, phone);
+	strncpy(ibuf, phone, sizeof(ibuf) - 1);
 	for (p = ibuf, q = obuf; *p != '\0'; p++) {
 		if (*p != ' ') {
 			*q = *p;
@@ -194,10 +197,10 @@
 		}
 	}
 	*q = '\0';
-	strcpy(ibuf, obuf);
+	strncpy(ibuf, obuf, sizeof(ibuf) - 1);
 
 	remove_parens(ibuf, obuf);
-	strcpy(ibuf, obuf);
+	strncpy(ibuf, obuf, sizeof(ibuf) - 1);
 
 	/* Look for "+" - if followed by a number,
 	   assume it's an international number and leave
@@ -213,7 +216,7 @@
 
 		/* Apply local munges */
 		munge_phone(ibuf, obuf);
-		strcpy(ibuf, obuf);
+		strncpy(ibuf, obuf, sizeof(ibuf) - 1);
 	}
 
 	/* Convert string of form abcd to remote-printer@d.c.b.a.tpc.int */
@@ -223,17 +226,19 @@
 		*q++ = '.';
 	}
 	*q = '\0';
-	strcpy(ibuf, obuf);
-	strcpy(obuf, "remote-printer");
+	strncpy(ibuf, obuf, sizeof(ibuf) - 1);
+	strncpy(obuf, "remote-printer", sizeof(obuf) - 1);
 
 	/* include userinfo if present */
 	if (userinfo != NULL && strlen(userinfo)) {
-		strcat(obuf, ".");
-		strcat(obuf, userinfo);
+		strncat(obuf, ".", sizeof(obuf) - 1);
+		strncat(obuf, userinfo, sizeof(obuf) - 1);
 	}
-	strcat(obuf, "@");
-	strcat(obuf, ibuf);		/* tack on reversed phone number */
-	strcat(obuf, TPCDOMAIN);	/* tack on domain name */
+	strncat(obuf, "@", sizeof(obuf) - strlen(obuf) - 1);
+	/* tack on reversed phone number */
+	strncat(obuf, ibuf, sizeof(obuf) - strlen(obuf) - 1);
+	/* tack on domain name */
+	strncat(obuf, TPCDOMAIN, sizeof(obuf) - strlen(obuf) - 1);
 	p = strdup(obuf);
 	return(p);
 }
--- openldap-2.0.12/clients/mail500/main.c	Sun Aug 20 22:40:25 2000
+++ openldap-2.0.12/clients/mail500/main.c	Fri Aug 31 15:42:30 2001
@@ -240,11 +240,12 @@
 		char	buf[1024];
 		int	i;
 
+		memset( buf, '\0', sizeof(buf) );
 		syslog( LOG_ALERT, "running as %d", geteuid() );
-		strcpy( buf, argv[0] );
+		strncpy( buf, argv[0], sizeof(buf) - 1 );
 		for ( i = 1; i < argc; i++ ) {
-			strcat( buf, " " );
-			strcat( buf, argv[i] );
+			strncat( buf, " ", sizeof(buf) - 1 );
+			strncat( buf, argv[i], sizeof(buf) - strlen(buf) - 1 );
 		}
 
 		syslog( LOG_ALERT, "args: (%s)", buf );
@@ -1068,10 +1069,11 @@
 		char	buf[1024];
 		int	i;
 
-		strcpy( buf, to[0] );
+		memset( buf, '\0', sizeof(buf) );
+		strncpy( buf, to[0], sizeof(buf) - 1 );
 		for ( i = 1; to[i] != NULL; i++ ) {
-			strcat( buf, " " );
-			strcat( buf, to[i] );
+			strncat( buf, " ", sizeof(buf) - strlen(buf) - 1 );
+			strncat( buf, to[i], sizeof(buf) - strlen(buf) - 1 );
 		}
 
 		syslog( LOG_ALERT, "send_message execing sendmail: (%s)", buf );
@@ -1126,10 +1128,11 @@
 			char	buf[1024];
 			int	i;
 
-			strcpy( buf, argv[0] );
+			memset( buf, '\0', sizeof(buf) );
+			strncpy( buf, argv[0], sizeof(buf) - 1 );
 			for ( i = 1; i < argc; i++ ) {
-				strcat( buf, " " );
-				strcat( buf, argv[i] );
+				strncat( buf, " ", sizeof(buf) - strlen(buf) - 1 );
+				strncat( buf, argv[i], sizeof(buf) - strlen(buf) - 1 );
 			}
 
 			syslog( LOG_ALERT, "execing sendmail: (%s)", buf );
@@ -1165,6 +1168,8 @@
 	WAITSTATUSTYPE	status;
 #endif
 
+	memset( buf, '\0', sizeof(buf) );
+
 	if ( strcmp( MAIL500_BOUNCEFROM, mailfrom ) == 0 ) {
 	    mailfrom = errorsfrom;
 	}
@@ -1181,10 +1186,10 @@
 	if ( debug ) {
 		int	i;
 
-		strcpy( buf, argv[0] );
+		strncpy( buf, argv[0], sizeof(buf) - 1 );
 		for ( i = 1; argv[i] != NULL; i++ ) {
-			strcat( buf, " " );
-			strcat( buf, argv[i] );
+			strncat( buf, " ", sizeof(buf) - strlen(buf) - 1 );
+			strncat( buf, argv[i], sizeof(buf) - strlen(buf) - 1 );
 		}
 
 		syslog( LOG_ALERT, "execing sendmail: (%s)", buf );
--- openldap-2.0.12/clients/rcpt500/main.c	Thu Aug 17 22:32:52 2000
+++ openldap-2.0.12/clients/rcpt500/main.c	Fri Aug 31 15:42:30 2001
@@ -343,6 +343,7 @@
     char	*s, *p;
     static char argbuf[ MAXSIZE ];
 
+    memset( argbuf, '\0', sizeof(argbuf) );
     p = text;
     for ( s = argbuf; *p != '\0'; ++p ) {
 	*s++ = TOLOWER( (unsigned char) *p );
@@ -352,7 +353,7 @@
     for ( i = 0; rcpt_cmds[ i ].cmd_text != NULL; ++i ) {
 	if (( s = strstr( argbuf, rcpt_cmds[ i ].cmd_text )) != NULL
 	    && isspace( (unsigned char) s[ strlen( rcpt_cmds[ i ].cmd_text ) ] )) {
-	    strcpy( argbuf, text + (s - argbuf) + strlen( rcpt_cmds[ i ].cmd_text ));
+	    strncpy( argbuf, text + (s - argbuf) + strlen( rcpt_cmds[ i ].cmd_text ), sizeof(argbuf) - 1);
 	    *argp = argbuf;
 	    while ( isspace( (unsigned char) **argp )) {
 		++(*argp);
--- openldap-2.0.12/clients/rcpt500/query.c	Sat Jul 21 15:01:37 2001
+++ openldap-2.0.12/clients/rcpt500/query.c	Fri Aug 31 15:42:30 2001
@@ -246,7 +246,7 @@
 static int
 append_text( void *reply, char *text, ber_len_t len )
 {
-    strcat( (char *) reply, text );
+    strncat( (char *) reply, text, len );
     return( len );
 }
     
--- openldap-2.0.12/contrib/saucer/main.c	Tue Jun 13 13:57:06 2000
+++ openldap-2.0.12/contrib/saucer/main.c	Fri Aug 31 15:42:30 2001
@@ -239,8 +239,8 @@
 						default_dn[0] = 0;
 						for (rdn = exploded_dn + 1; *rdn; rdn++) {
 							if (default_dn[0])
-								strcat(default_dn, ", ");
-							strcat(default_dn, *rdn);
+								strncat(default_dn, ", ", sizeof(default_dn) - strlen(default_dn) - 1);
+							strncat(default_dn, *rdn, sizeof(default_dn) - strlen(default_dn) - 1);
 						}
 					}
 					ldap_value_free(exploded_dn);
@@ -251,9 +251,9 @@
 					if (relative  &&  !is_whitespace(default_dn)) {
 						char	buf[DN_MAXLEN];
 
-						strcpy(default_dn, strcat(strcat(strcpy(buf, dn), ", "), default_dn));
+						strncpy(default_dn, strcat(strncat(strncpy(buf, dn, sizeof(buf) - 1), ", ", sizeof(buf) - strlen(buf) - 1), default_dn), sizeof(default_dn) - 1);
 					} else
-						strcpy(default_dn, dn);
+						strncpy(default_dn, dn, sizeof(default_dn) - 1);
 					ldap_value_free(exploded_dn);
 				} else
 					puts("Invalid distinguished name.");
@@ -672,7 +672,7 @@
 	if (!dn[0])
 		return default_dn;
 
-	return strcat(strcat(strcpy(dn_buf, dn), ", "), default_dn);
+	return strncat(strncat(strncpy(dn_buf, dn, sizeof(dn_buf) - 1), ", ", sizeof(dn_buf) - strlen(dn_buf) - 1), default_dn, sizeof(dn_buf) - strlen(dn_buf) - 1);
 }
 
 void show_syntax(unsigned int cmdnum)
@@ -735,17 +735,18 @@
 
 	rcfile[0] = 0;
 
+	memset(rcfile, '\0', sizeof(rcfile));
 #ifdef unix
 	{
 #include <pwd.h>
 		struct passwd	*pwent;
 
 		if (pwent = getpwuid(getuid()))
-			strcat(strcpy(rcfile, pwent->pw_dir), "/");
-		strcat(rcfile, ".saucerrc");
+			strncat(strncpy(rcfile, pwent->pw_dir, sizeof(rcfile) - 1), "/", sizeof(rcfile) - strlen(rcfile) - 1);
+		strncat(rcfile, ".saucerrc", sizeof(rcfile) - strlen(rcfile) - 1);
 	}
 #else
-	strcpy(rcfile, "saucer.rc");
+	strncpy(rcfile, "saucer.rc", sizeof(rcfile) - 1);
 #endif
 
 	return fopen(rcfile, "r");
--- openldap-2.0.12/libraries/libldap/test.c	Sat Jul 21 15:01:40 2001
+++ openldap-2.0.12/libraries/libldap/test.c	Fri Aug 31 15:42:30 2001
@@ -250,7 +250,7 @@
 #endif /* LDAP_API_FEATURE_X_OPENLDAP_V2_KBIND */
 
 		getline( dn, sizeof(dn), stdin, "re-bind dn? " );
-		strcat( dn, dnsuffix );
+		strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 
 	if ( authmethod == LDAP_AUTH_SIMPLE && dn[0] != '\0' ) {
 			getline( passwd, sizeof(passwd), stdin,
@@ -372,7 +372,7 @@
 			switch ( command2 ) {
 			case 'd':	/* add */
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				if ( (attrs = get_modlist( NULL, "attr? ",
 				    "value? " )) == NULL )
 					break;
@@ -405,7 +405,7 @@
 			method = LDAP_AUTH_SIMPLE;
 #endif /* LDAP_API_FEATURE_X_OPENLDAP_V2_KBIND */
 			getline( dn, sizeof(dn), stdin, "dn? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) -1 );
 
 			if ( method == LDAP_AUTH_SIMPLE && dn[0] != '\0' )
 				getline( passwd, sizeof(passwd), stdin,
@@ -435,7 +435,7 @@
 			method = LDAP_AUTH_SIMPLE;
 #endif /* LDAP_API_FEATURE_X_OPENLDAP_V2_KBIND */
 			getline( dn, sizeof(dn), stdin, "dn? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 
 			if ( dn[0] != '\0' )
 				getline( passwd, sizeof(passwd), stdin,
@@ -455,7 +455,7 @@
 
 		case 'c':	/* compare */
 			getline( dn, sizeof(dn), stdin, "dn? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 			getline( attr, sizeof(attr), stdin, "attr? " );
 			getline( value, sizeof(value), stdin, "value? " );
 
@@ -500,7 +500,7 @@
 		case 'm':	/* modify or modifyrdn */
 			if ( strncmp( line, "modify", 4 ) == 0 ) {
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				if ( (mods = get_modlist(
 				    "mod (0=>add, 1=>delete, 2=>replace -1=>done)? ",
 				    "attribute type? ", "attribute value? " ))
@@ -513,7 +513,7 @@
 					    id );
 			} else if ( strncmp( line, "modrdn", 4 ) == 0 ) {
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				getline( rdn, sizeof(rdn), stdin, "newrdn? " );
 				if ( (id = ldap_modrdn( ld, dn, rdn )) == -1 )
 					ldap_perror( ld, "ldap_modrdn" );
@@ -558,7 +558,7 @@
 
 			case 'm':	/* remove */
 				getline( dn, sizeof(dn), stdin, "dn? " );
-				strcat( dn, dnsuffix );
+				strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 				if ( (id = ldap_delete( ld, dn )) == -1 )
 					ldap_perror( ld, "ldap_delete" );
 				else
@@ -574,7 +574,7 @@
 
 		case 's':	/* search */
 			getline( dn, sizeof(dn), stdin, "searchbase? " );
-			strcat( dn, dnsuffix );
+			strncat( dn, dnsuffix, sizeof(dn) - strlen(dn) - 1 );
 			getline( line, sizeof(line), stdin,
 			    "scope (0=Base, 1=One Level, 2=Subtree)? " );
 			scope = atoi( line );
