diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/fax500/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/fax500/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/fax500/Makefile.in	2000-06-13 13:57:01.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/fax500/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -37,7 +37,7 @@
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	@(								\
 	    for prg in $(EXEPRGS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/finger/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/finger/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/finger/Makefile.in	2000-06-13 13:57:01.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/finger/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -24,7 +24,7 @@
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/gopher/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/gopher/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/gopher/Makefile.in	2000-06-13 13:57:01.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/gopher/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -36,7 +36,7 @@
 	-$(MKDIR) $(DESTDIR)$(libexecdir) $(DESTDIR)$(datadir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/mail500/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/mail500/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/mail500/Makefile.in	2000-06-13 13:57:02.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/mail500/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -24,7 +24,7 @@
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/maildap/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/maildap/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/maildap/Makefile.in	2000-06-13 13:57:02.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/maildap/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -24,7 +24,7 @@
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/rcpt500/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/rcpt500/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/rcpt500/Makefile.in	2000-06-13 13:57:03.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/rcpt500/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -25,7 +25,7 @@
 	-$(MKDIR) $(DESTDIR)$(libexecdir) $(DESTDIR)$(datadir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/tools/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/tools/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/tools/Makefile.in	2000-07-04 13:58:44.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/tools/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -61,7 +61,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/clients/ud/Makefile.in openldap-2.2.13/openldap-2.0.27/clients/ud/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/clients/ud/Makefile.in	2001-10-10 12:41:32.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/clients/ud/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -29,7 +29,7 @@
 	-mv -f $(DESTDIR)$(bindir)/ud $(DESTDIR)$(bindir)/ud-
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.0.27/servers/slapd/Makefile.in openldap-2.2.13/openldap-2.0.27/servers/slapd/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/servers/slapd/Makefile.in	2001-07-21 22:28:28.000000000 -0400
+++ openldap-2.2.13/openldap-2.0.27/servers/slapd/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -232,7 +232,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)
-	$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@if test ! -z "$(SLAPD_MODULES)" ; then \
 	    for i in back-* shell-backends tools; do \
diff -urN openldap-2.2.13.orig/openldap-2.0.27/servers/slapd/tools/Makefile.in openldap-2.2.13/openldap-2.0.27/servers/slapd/tools/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/servers/slapd/tools/Makefile.in	2002-01-04 15:38:36.000000000 -0500
+++ openldap-2.2.13/openldap-2.0.27/servers/slapd/tools/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -83,7 +83,7 @@
 install-ldbm-yes install-ldbm-mod: FORCE
 	-$(MKDIR) $(DESTDIR)$(sbindir)
 	@for bin in $(PROGRAMS); do \
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 			$$bin$(EXEEXT) $(DESTDIR)$(sbindir); \
 	done
 
diff -urN openldap-2.2.13.orig/openldap-2.0.27/servers/slurpd/Makefile.in openldap-2.2.13/openldap-2.0.27/servers/slurpd/Makefile.in
--- openldap-2.2.13.orig/openldap-2.0.27/servers/slurpd/Makefile.in	2002-01-04 15:38:36.000000000 -0500
+++ openldap-2.2.13/openldap-2.0.27/servers/slurpd/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -48,7 +48,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.1.30/clients/tools/Makefile.in openldap-2.2.13/openldap-2.1.30/clients/tools/Makefile.in
--- openldap-2.2.13.orig/openldap-2.1.30/clients/tools/Makefile.in	2003-03-29 10:45:43.000000000 -0500
+++ openldap-2.2.13/openldap-2.1.30/clients/tools/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -89,7 +89,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.1.30/servers/slapd/Makefile.in openldap-2.2.13/openldap-2.1.30/servers/slapd/Makefile.in
--- openldap-2.2.13.orig/openldap-2.1.30/servers/slapd/Makefile.in	2003-03-31 11:48:17.000000000 -0500
+++ openldap-2.2.13/openldap-2.1.30/servers/slapd/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -317,7 +317,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)
-	$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in back-* shell-backends tools; do \
 	    if test -d $$i -a -f $$i/Makefile ; then \
diff -urN openldap-2.2.13.orig/openldap-2.1.30/servers/slapd/tools/Makefile.in openldap-2.2.13/openldap-2.1.30/servers/slapd/tools/Makefile.in
--- openldap-2.2.13.orig/openldap-2.1.30/servers/slapd/tools/Makefile.in	2003-02-09 11:31:39.000000000 -0500
+++ openldap-2.2.13/openldap-2.1.30/servers/slapd/tools/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -77,7 +77,7 @@
 install-local: FORCE
 	-$(MKDIR) $(DESTDIR)$(sbindir)
 	@for bin in $(PROGRAMS); do \
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 			$$bin$(EXEEXT) $(DESTDIR)$(sbindir); \
 	done
 
diff -urN openldap-2.2.13.orig/openldap-2.1.30/servers/slurpd/Makefile.in openldap-2.2.13/openldap-2.1.30/servers/slurpd/Makefile.in
--- openldap-2.2.13.orig/openldap-2.1.30/servers/slurpd/Makefile.in	2003-12-17 22:25:14.000000000 -0500
+++ openldap-2.2.13/openldap-2.1.30/servers/slurpd/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -61,7 +61,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.2.13/clients/tools/Makefile.in openldap-2.2.13/openldap-2.2.13/clients/tools/Makefile.in
--- openldap-2.2.13.orig/openldap-2.2.13/clients/tools/Makefile.in	2004-01-01 13:16:26.000000000 -0500
+++ openldap-2.2.13/openldap-2.2.13/clients/tools/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -98,7 +98,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.2.13/servers/slapd/Makefile.in openldap-2.2.13/openldap-2.2.13/servers/slapd/Makefile.in
--- openldap-2.2.13.orig/openldap-2.2.13/servers/slapd/Makefile.in	2006-01-28 11:53:26.000000000 -0500
+++ openldap-2.2.13/openldap-2.2.13/servers/slapd/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -355,7 +355,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i -a -f $$i/Makefile ; then \
diff -urN openldap-2.2.13.orig/openldap-2.2.13/servers/slapd/Makefile.in.pie openldap-2.2.13/openldap-2.2.13/servers/slapd/Makefile.in.pie
--- openldap-2.2.13.orig/openldap-2.2.13/servers/slapd/Makefile.in.pie	2004-04-12 14:07:40.000000000 -0400
+++ openldap-2.2.13/openldap-2.2.13/servers/slapd/Makefile.in.pie	2006-01-28 12:35:49.000000000 -0500
@@ -355,7 +355,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i -a -f $$i/Makefile ; then \
diff -urN openldap-2.2.13.orig/openldap-2.2.13/servers/slapd/Makefile.in.toollinks openldap-2.2.13/openldap-2.2.13/servers/slapd/Makefile.in.toollinks
--- openldap-2.2.13.orig/openldap-2.2.13/servers/slapd/Makefile.in.toollinks	2006-01-28 11:53:26.000000000 -0500
+++ openldap-2.2.13/openldap-2.2.13/servers/slapd/Makefile.in.toollinks	2006-01-28 12:35:49.000000000 -0500
@@ -355,7 +355,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i -a -f $$i/Makefile ; then \
diff -urN openldap-2.2.13.orig/openldap-2.2.13/servers/slurpd/Makefile.in openldap-2.2.13/openldap-2.2.13/servers/slurpd/Makefile.in
--- openldap-2.2.13.orig/openldap-2.2.13/servers/slurpd/Makefile.in	2006-01-28 11:53:26.000000000 -0500
+++ openldap-2.2.13/openldap-2.2.13/servers/slurpd/Makefile.in	2006-01-28 12:35:49.000000000 -0500
@@ -60,7 +60,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
diff -urN openldap-2.2.13.orig/openldap-2.2.13/servers/slurpd/Makefile.in.pie openldap-2.2.13/openldap-2.2.13/servers/slurpd/Makefile.in.pie
--- openldap-2.2.13.orig/openldap-2.2.13/servers/slurpd/Makefile.in.pie	2004-01-01 13:16:42.000000000 -0500
+++ openldap-2.2.13/openldap-2.2.13/servers/slurpd/Makefile.in.pie	2006-01-28 12:35:49.000000000 -0500
@@ -60,7 +60,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
